core/src/main/java/com/linecorp/armeria/internal/client/DefaultDnsNameResolver.java

```diff
@@ -96,12 +89,22 @@ public void operationComplete(Future<List<DnsRecord>> future) throws Exception {
 
                 if (--remaining == 0) {
                     if (!records.isEmpty()) {
+                        if (records.size() > 1) {
+                            final List<DnsRecordType> preferredOrder =
```

Comment:
I think we always have a fixed set of types, so since this is an internal class we could pass them to the constructor and only do this once. Can also use Ordering.explicit